### PR TITLE
feat!: Replace random password generation with manage master password, min AWS provider version increased to 5.0, `id` output replaced with `identifier`

### LINKED ISSUE #488: Support for storing master user password in Secrets Manager
Copied from the provider issue https://github.com/hashicorp/terraform-provider-aws/issues/28538, supported since [provider ver. 4.61.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#4610-march-30-2023):

> Amazon RDS now supports integration with AWS Secrets Manager to streamline how you manage your master user password for your RDS database instances. With this feature, RDS fully manages the master user password and stores it in AWS Secrets Manager whenever your RDS database instances are created, modified, or restored. The new feature supports the entire lifecycle maintenance for your RDS master user password including regular and automatic password rotations; removing the need for you to manage rotations using custom Lambda functions.
> 
> RDS integration with AWS Secrets Manager improves your database security by ensuring your RDS master user password is not visible in plaintext to administrators or engineers during your database creation workflow. Furthermore, you have flexibility in encrypting the secrets using your own managed key or by using a KMS key provided by AWS Secrets Manager. RDS and AWS Secrets Manager provide you the ease and security in managing your master user password for your database instances, relieving you from complex credential management activities such as setting up custom Lambda functions to manage password rotations.

This is a great improvement from the security perspective, I would love to see it propagated to this module.

### LINKED ISSUE #495: add identifier to outputs
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality
aws provider: 5.0.0

## Is your request related to a problem? Please describe.
id is no longer the AWS database identifier - id is now the dbi-resource-id. Refer to identifier instead of id to use the database's identifier

## Describe the solution you'd like.
add to module outputs db identifier value

### LINKED ISSUE #498: db_instance_id is returning the DB Resource ID
## Description

I am using this module as part of a module in our project's infrastructure. I recently began adding some automated monitoring and was trying to set up some alarms for some RDS metrics. When I attempted to have the module output the `db_instance_id`, I instead get the `db_resource_id` returned. 

The console indicates that an appropriate identifier is set, and requesting the arn for the rds instance shows the identifier correctly. 

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 5.9.0

- Terraform version: 1.4.6

- Provider version(s): 5.1.0

## Reproduction Code [Required]
```
output "db_instance_id" {
  value = module.db.db_instance_id
}
```

1. Deploy RDS using module
2. Request `db_instance_id` of the deployed database

## Expected behavior
`db_instance_id` should return the instance identifier and not the resource identifier.

## Actual behavior
`db_instance_id` returns the resource identifier.

### Terminal Output Screenshot(s)
Example (partially redacted):
```
app_db_id = "db-63B5TPHO4P6G2NRAVXEEXXXXXX"
oauth_db_id = "db-IC6E5RWM5UD6OOYMFTKIXXXXXX"
```

## Additional context
Note that another infrastructure deployment operating with provider version 4.58.0 and module version 5.6.0 returns the instance identifier correctly.

### PR DESCRIPTION
**Pull request by decayofmind**
## Description

- Replace random password generation with new RDS manage master user password which is now enabled by default
- Increase AWS provider minimum supported version to 5.0
- Replace `id` output for DB instance to `identifier` (ref: https://github.com/hashicorp/terraform-provider-aws/pull/31232)
- Password outputs have been removed 

## Motivation and Context

Support of this new feature allows users to offload master password lifecycle completely to AWS, including rotation and avoid storing credentials in state file.

- Resolves #488
- Resolves #494
- Resolves #495 
- Resolves #496
- Resolves #498
- Resolves #499 

## Breaking Changes
- Yes

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -84,12 +84,14 @@ module "db_instance" {
 
   db_name                             = var.db_name
   username                            = var.username
-  password                            = local.password
+  password                            = !var.manage_master_user_password ? local.password : null
```
Comment (gruckion) [inline main.tf:87]:
```suggestion
  password                            = var.manage_master_user_password ? null : local.password
```

Reads better

**Comment by bryantbiggs** [comment]
similar to what we did with the Aurora module, we'll want to remove the random password generation here https://github.com/terraform-aws-modules/terraform-aws-rds-aurora/pull/335

This will be a breaking change but its the proper solution for keeping secrets secure, out of Terraform state, and allow them to more easily be rotated without Terraform interference

**Comment by asinghggt** [comment]
any update on this?

**Comment by rlmartin** [comment]
Any updates on this?

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 6.0.0 :tada:

**Pull request merged**